### DISCUSSION TITLE
`trial.suggest_float()` : absolute value

### DISCUSSION THREAD

**Discussion by GregoireHein** [optuna/optuna#5406] [Q&A]
Hi, I'm looking for a way to define a contraint on the absolute value of a suggested float.
For example, I can specify `1 <= x1 <= 10` using `x1 = trial.suggest_float("x1", 1, 10)`.

How can I specify `1 <= abs(x1) <= 10` ?

**Comment by THB0705** [comment]
Create an additional proxy variable?

`x_sign = trial.suggest_int("x_sign",0,1)`
`x_magnitude = trial.suggest_float("x_magnitude",1,10)`
`x1 = x_sign*x_magnitude`

It's janky, but could work depending on your application.
